ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -1167,13 +1210,28 @@ private StoreFindToken findEntriesFromSegmentStartOffset(Offset initialSegmentSt
   /**
    * We can have duplicate entries in the message entries since updates can happen to the same key. For example,
    * insert a key followed by a delete. This would create two entries in the journal or the index. A single findInfo
-   * could read both the entries. The findInfo should return as clean information as possible. This method removes
-   * the oldest duplicate in the list.
+   * could read both the entries. The findInfo should return as clean information as possible.
+   * <p/>
+   * This function choose a delete entry over all entries but chooses a put entry over a ttl update entry
    * @param messageEntries The message entry list where duplicates need to be removed
    */
   private void eliminateDuplicates(List<MessageInfo> messageEntries) {
-    Set<StoreKey> setToFindDuplicate = new HashSet<StoreKey>();
-    ListIterator<MessageInfo> messageEntriesIterator = messageEntries.listIterator(messageEntries.size());
+    Set<StoreKey> setToFindDuplicate = new HashSet<>();
```

Comment:
This could probably be done the same way as my other comment. Basically, use `List::removeIf()` and exploit the return boolean of `Set::add()`.

Reply:
Will take a look - but what's the other related comment? May have missed it

Reply:
https://github.com/linkedin/ambry/pull/937#discussion_r193594025

Reply:
Made the change here too